cmake/public/gridtools_setup_targets.cmake

```diff
@@ -0,0 +1,165 @@
+set(_gt_gridtools_setup_targets_dir ${CMAKE_CURRENT_LIST_DIR})
```

Comment:
This file is largely copied from the main CMakeLists.txt. I added support for dealing with the namespace when used from the installation. I tried to document why as this is non-standard practice.